test/trim_trailing_whitespaces_test.rb

```diff
@@ -0,0 +1,195 @@
+        out.text("#{whitespace}a#{add_trailing ? whitespace : ''}")
+        out.break
+        out.text("#{whitespace}b")
+      }
+      expected = <<~LANG.chomp
+        #{whitespace}a
+        #{whitespace}b
+      LANG
+
+      trim_trailing_whitespaces_round_trip(builder_block, expected, whitespace)
+    end
+
+    it "trims multiple `#{whitespace}` trailing whitespaces" do
+      builder_block = proc { |out, add_trailing = true|
+        out.text("#{whitespace}a#{add_trailing ? whitespace * 42 : ''}")
+        out.break
+        out.text("#{whitespace}b")
+      }
+      expected = <<~LANG.chomp
+        #{whitespace}a
+        #{whitespace}b
+      LANG
+
+      trim_trailing_whitespaces_round_trip(builder_block, expected, whitespace)
+    end
+
+    it "trims `#{whitespace}` whitespace of previous parent group" do
+      builder_block = proc { |out, add_trailing = true|
+        out.text("a#{add_trailing ? whitespace : ''}")
+        out.group {
+          out.group {
+            out.break
+            out.text('b')
+          }
+        }
+      }
+      expected = <<~LANG.chomp
+        a
+        b
+      LANG
+
+      trim_trailing_whitespaces_round_trip(builder_block, expected, whitespace)
+    end
+
+    it "trims `#{whitespace}` whitespace of previous child group" do
+      builder_block = proc { |out, add_trailing = true|
+        out.group {
+          out.group {
+            out.group {
+              out.text("a#{add_trailing ? whitespace : ''}")
+            }
+          }
+          out.break
+          out.text('b')
+        }
+      }
+      expected = <<~LANG.chomp
+        a
+        b
+      LANG
+
+      trim_trailing_whitespaces_round_trip(builder_block, expected, whitespace)
+    end
+
+    it "trims `#{whitespace}` of pattern: text, whitespace(s), break" do
+      builder_block1 = proc { |out, add_trailing = true|
+        out.text("#{whitespace}a#{add_trailing ? whitespace : ''}")
+        out.break
+      }
+      builder_block2 = proc { |out, add_trailing = true|
+        out.text("#{whitespace}a#{add_trailing ? whitespace * 42 : ''}")
+        out.break
+      }
+      expected = <<~LANG
+        #{whitespace}a
+      LANG
+
+      trim_trailing_whitespaces_round_trip(builder_block1, expected, whitespace)
+      trim_trailing_whitespaces_round_trip(builder_block2, expected, whitespace)
+    end
+
+    it "trims `#{whitespace}` of pattern: text, whitespace(s), breaks" do
+      builder_block1 = proc { |out, add_trailing = true|
+        out.text("#{whitespace}a#{add_trailing ? whitespace : ''}")
+        out.break
+        out.break
+        out.break
+        out.break
+        out.break
+      }
+      builder_block2 = proc { |out, add_trailing = true|
+        out.text("#{whitespace}a#{add_trailing ? whitespace * 42 : ''}")
+        out.break
+        out.break
+        out.break
+        out.break
+        out.break
+      }
+      expected = <<~LANG
+        #{whitespace}a
+
+
+
+
+      LANG
```

Comment:
I find these here-docs hard to read.

I fail to see how this would be more readable than
```ruby
expected = "#{whitespace}a}\n\n\n\n\n"
```